compiler/noirc_frontend/src/tests.rs

```diff
@@ -1430,16 +1430,17 @@ fn specify_method_types_with_turbofish() {
         }
         
         impl<T> Foo<T> {
-            fn generic_method<U>(_self: Self) where U: Default {
+            fn generic_method<U>(_self: Self) -> U where U: Default {
```

Comment:
We had a mismatch in the expected and actual type returned here now that `Self` for the trait method actually is bound to `U`